.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,454 @@
+          cmake --build build_examples -j
+
+      - name: Test Tutorials
+        run: |
+          # We need to execute the HelloWorld in the same folder as the PCAP file
+          cp build_examples/Tutorials/Tutorial-HelloWorld/Tutorial-HelloWorld Examples/Tutorials/Tutorial-HelloWorld/
+          cd Examples/Tutorials/Tutorial-HelloWorld && ./Tutorial-HelloWorld
+
+  dpdk:
+    runs-on: ubuntu-20.04
+    container: seladb/${{ matrix.image }}
+    strategy:
+      matrix:
+        image:
+          [
+            ubuntu2004-dpdk2111,
+            ubuntu2004-dpdk2011,
+            ubuntu2004-dpdk1911,
+            ubuntu2004-dpdk1811,
+          ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Mark repository as safe (Checkout actions is not run with same UID)
+        run: |
+          git config --global --add safe.directory /__w/PcapPlusPlus/PcapPlusPlus
+          git config --global --add safe.directory ${{ github.workspace }}
+
+      - name: Configure PcapPlusPlus
+        run: cmake -DPCAPPP_USE_DPDK=ON -S . -B $BUILD_DIR
+
+      - name: Build PcapPlusPlus
+        run: cmake --build $BUILD_DIR -j
+
+      - name: Check AVX
+        run: grep avx /proc/cpuinfo
+
+      - name: Test Packet++
+        run: |
+          cd Tests/Packet++Test
+          if [ -n "$(grep avx512 /proc/cpuinfo)" ]; then Bin/Packet++Test; else echo AVX-512 SUPPORT NOT FOUND, CANNOT RUN Packet++Test; fi
+
+      - name: Test Pcap++
+        run: |
+          cd Tests/Pcap++Test
+          if [ -n "$(grep avx512 /proc/cpuinfo)" ]; then Bin/Pcap++Test -n; else echo AVX-512 SUPPORT NOT FOUND, CANNOT RUN Pcap++Test; fi
+
+  pfring:
+    runs-on: ubuntu-20.04
+    container: seladb/ubuntu2004-pfring
+    strategy:
+      matrix:
+        include:
+          - configure: cmake -DPCAPPP_USE_PF_RING=ON -DPF_RING_ROOT="/PF_RING" -S . -B $BUILD_DIR
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Mark repository as safe (Checkout actions is not run with same UID)
+        run: |
+          git config --global --add safe.directory /__w/PcapPlusPlus/PcapPlusPlus
+          git config --global --add safe.directory ${{ github.workspace }}
+
+      - name: Configure PcapPlusPlus
+        run: ${{ matrix.configure }}
+
+      - name: Build PcapPlusPlus
+        run: cmake --build $BUILD_DIR -j
+
+      - name: Test Packet++
+        run: |
+          cd Tests/Packet++Test
+          Bin/Packet++Test
+
+      - name: Test Pcap++
+        run: |
+          cd Tests/Pcap++Test
+          Bin/Pcap++Test -n
+
+  macos:
+    runs-on: ${{ matrix.os-version }}
+    strategy:
+      matrix:
+        os-version: [macos-11, macos-12]
+        arch: [x86_64, arm64]
+        # Handle ZSTD build by Cirrus CI
+        # config-zstd: [ON, OFF]
```

Comment:
You turned these off but didn't add these tests in `.cirrus.yml` 🤔

Reply:
done

Reply:
I see all of them fail because of the version issue: https://github.com/seladb/PcapPlusPlus/pull/944#discussion_r969531788 😕 

```
CMake Error at CMakeLists.txt:17 (project):
  VERSION "22.05+" format invalid.
```